src/common/turbo.ts

```diff
@@ -47,31 +59,66 @@ export async function signedRequestHeadersFromSigner({
   signer: TurboArNSSigner;
   nonce?: string;
 }): Promise<RawAxiosRequestHeaders> {
-  await (signer as ArconnectSigner).setPublicKey?.();
-  const signature = await signer.sign(Uint8Array.from(Buffer.from(nonce)));
-
-  let publicKey: string;
-  switch (signer.signatureType) {
-    case SignatureConfig.ARWEAVE:
-      publicKey = toB64Url(signer.publicKey);
-      break;
-    case SignatureConfig.ETHEREUM:
-      publicKey = '0x' + signer.publicKey.toString('hex');
-      break;
-    // TODO: solana sig support
-    // case SignatureConfig.SOLANA:
-    // case SignatureConfig.ED25519:
-    default:
-      throw new Error(
-        `Unsupported signer type for signing requests: ${signer.signatureType}`,
-      );
+  let signature: string | undefined = undefined;
+  let publicKey: string | undefined = undefined;
+
+  const signatureType = isWanderArweaveBrowserSigner(signer)
+    ? SignatureConfig.ARWEAVE
+    : (signer as Signer).signatureType;
+
+  if (signer instanceof ArconnectSigner) {
+    signature = toB64Url(
+      Buffer.from(
+        await signer['signer'].signMessage(Uint8Array.from(Buffer.from(nonce))),
+      ),
+    );
   }
 
+  // equivalent to window.arweaveWallet
+  if (isWanderArweaveBrowserSigner(signer)) {
+    signature = toB64Url(
+      Buffer.from(
+        await signer.signMessage(Uint8Array.from(Buffer.from(nonce))),
+      ),
+    );
+  } else if (
+    signer instanceof ArweaveSigner ||
+    signer instanceof EthereumSigner ||
+    signer instanceof InjectedEthereumSigner
+  ) {
```

Comment:
isn't this the as `isTurboArNSSigner`?

Reply:
No - isTurboArNSSigner includes browser signer which doesnt implement the Signer interface from arbundles.